features/step_definitions/affordability.rb

```diff
@@ -284,55 +284,24 @@
 end
 
 Then(/^I should see the correct results for a single applicant$/) do
-  expect(step_three.repayments).to have_text('£1,706.21')
-  expect(step_three.risk_chart).to have_text('53%')
-  expect(step_three.essentials).to have_text('53%')
-  expect(step_three.essentials).to have_text('£3,206.21')
-  expect(step_three.total_leftover).to have_text('47%')
-  expect(step_three.total_leftover).to have_text('£2,793.79')
-  expect(step_three.remaining).to have_text('£1,893.79')
-  expect(step_three.what_if_changes).to have_text('2%')
-  expect(step_three.what_if_changes).to have_text('£2,103.35')
-  expect(step_three.what_if_remaining).to have_text('£1,496.65')
-end
-
-Then(/^I should see the correct results for a single applicant for JavaScript$/) do
-  expect(step_three.repayments_js).to have_text('£1,706.21')
-  expect(step_three.risk_chart_js).to have_text('53%')
-  expect(step_three.essentials_js).to have_text('53%')
-  expect(step_three.essentials_js).to have_text('£3,206.21')
-  expect(step_three.total_leftover_js).to have_text('47%')
-  expect(step_three.total_leftover_js).to have_text('£2,793.79')
-  expect(step_three.remaining_js).to have_text('£1,893.79')
-  expect(step_three.what_if_changes_js).to have_text('3%')
-  expect(step_three.what_if_changes_js).to have_text('£2,318.20')
-  expect(step_three.what_if_remaining_js).to have_text('£1,281.80')
-end
-
+  expect(js_test? ? step_three.repayments_js : step_three.repayments).to have_text('£1,706.21')
+  expect(js_test? ? step_three.risk_chart_js : step_three.risk_chart).to have_text('53%')
+  expect(js_test? ? step_three.essentials_js : step_three.essentials).to have_text('53%')
+  expect(js_test? ? step_three.essentials_js : step_three.essentials).to have_text('£3,206.21')
+  expect(js_test? ? step_three.total_leftover_js : step_three.total_leftover).to have_text('47%')
+  expect(js_test? ? step_three.total_leftover_js : step_three.total_leftover).to have_text('£2,793.79')
+  expect(js_test? ? step_three.remaining_js : step_three.remaining).to have_text('£1,893.79')
+  expect(js_test? ? step_three.what_if_changes_js : step_three.what_if_changes).to have_text(js_test? ? '3%' : '2%')
```

Comment:
@Tr4pSt3R ok, I thought you might mean that but I didn't see what the contents of `js_test?` had to do with that.

Anyway, yes, it's a toss up. Either twice as many, almost identical lines or fewer lines with unsightly repeated inline logic.  I chose this way in the end because it's easier to see what's different between the two.

If anyone has a strong preference one over the other, I really don't myself so have no problem changing things if so.